src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -33,56 +33,113 @@ public PropertySubpatternCompletionProvider()
         {
         }
 
+        // Examples:
+        // is { $$
+        // is { Property.$$
+        // is { Property.Property2.$$
         public override async Task ProvideCompletionsAsync(CompletionContext context)
         {
             var document = context.Document;
             var position = context.Position;
             var cancellationToken = context.CancellationToken;
             var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
 
-            var token = TryGetOpenBraceOrCommaInPropertyPatternClause(tree, position, cancellationToken);
-            if (token == default || !(token.Parent.Parent is PatternSyntax))
+            var (propertyPatternClause, memberNameAccess) = TryGetPropertyPatternClause(tree, position, cancellationToken);
+            if (propertyPatternClause is null)
             {
                 return;
             }
 
             var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
-            var pattern = (PatternSyntax)token.Parent.Parent;
+            var pattern = (PatternSyntax)propertyPatternClause.Parent;
             var type = semanticModel.GetTypeInfo(pattern, cancellationToken).ConvertedType;
-            if (type == null)
+
+            if (memberNameAccess is not null)
+            {
+                // We have to figure out the type of the extended property ourselves, because
+                // the semantic model could not provide the answer we want in incomplete syntax:
+                // `c is { X. }`
+
+                type = GetMemberAccessType(type, memberNameAccess.Expression, document, semanticModel, position);
+            }
```

Comment:
i found this code confusing.  esp passing in the type to return a type.  could we just compute the type in whatever way is appropriate given the presence of a memberNameAccess or not?

Reply:
I didn't understand the suggestion.
In the example of `c is { X. }` we start with the type of `c` (given by `GetTypeInfo`), then we're going to access members (just `X` in this case) and we'll end up with the type of `c.X` as the starting point for completion.

Reply:
I've attemped a change, but I'm not sure it's what you had in mind.